app/__init__.py

```diff
@@ -34,106 +38,159 @@
         app.logger.error(e)
-        return jsonify(status=409, message=str(e))
+        return jsonify(status=errors.ERROR_STATUS, message=str(e))
 
 
 @app.route('/workspace/delete', methods=['POST'])
-def linchpin_delete_workspace():
+def linchpin_delete_workspace() -> Response:
+    """
+        POST request route for deleting workspaces.
+        RequestBody: {"name": "workspacename"}
+        :return : response with deleted workspace name and status
+    """
     try:
         data = request.json  # Get request body
         name = data["name"]
         # path specifying location of working directory inside server
-        for x in os.listdir(os.path.join(app.root_path + WORKING_DIR)):
+        for x in os.listdir(WORKING_PATH):
             if x == name:
-                shutil.rmtree(name)
+                shutil.rmtree(WORKING_PATH + "/" + name)
+                dbConn.db_remove(name)
                 return jsonify(name=name,
-                               status="Workspace deleted successfully")
-        return jsonify(status="Workspace " + name + " not found")
+                               status=response.DELETE_SUCCESS,
+                               mimetype='application/json')
+        return jsonify(status=response.NOT_FOUND)
+    except (KeyError, ValueError, TypeError):
+        return jsonify(status=errors.ERROR_STATUS,
+                       message=errors.KEY_ERROR_NAME)
     except Exception as e:
         app.logger.error(e)
-        return jsonify(status=409, message=str(e))
+        return jsonify(status=errors.ERROR_STATUS, message=str(e))
+
+
+def create_fetch_cmd(data) -> List[str]:
+    """
+        Creates a list to feed the subprocess in fetch API
+        :param data: JSON data from POST requestBody
+        :return a list for the subprocess to run
+    """
+    name = data['name']
+    url = data['url']
+    repo = None
+    # initial list
+    cmd = ["linchpin", "-w " + WORKING_DIR + name, "fetch"]
+
+    # Check for repoType field in request,
+    # Only true if it is set to web
+    if 'repoType' in data:
+        if data['repoType'] == 'web':
+            repo = 'web'
+            cmd.append("--web")
+
+    if 'rootfolder' in data:
+        cmd.extend(("--root", data['rootfolder']))
+
+    if repo is None and 'branch' in data:
+        cmd.extend(("--branch", data['branch']))
+
+    # last item to be added in the array
+    if 'url' in data:
+        cmd.append(str(url))
+    return cmd
 
 
 @app.route('/workspace/fetch', methods=['POST'])
-def linchpin_fetch_workspace():
+def linchpin_fetch_workspace() -> Response:
+    """
+        POST request route for fetching workspaces from a remote URL
+        RequestBody: {"name": "workspacename", "url": "www.github.com/someurl",
+        "rootfolder":"/path/to/folder"}
+        :return : response with fetched workspace name,status and code
+    """
     try:
         data = request.json  # Get request body
         name = data['name']
```

Comment:
@mansikulkarni96 How do you identify if i have multiple workspaces with same name. 
It would not allow me in first place. according to current logic. 
You would want to append id to workspace name before creating.
for example: 
if I give a request to create a workspace "testworkspace"
the workspace folder name would be --> 1_testworkspace 

Make sure you add status attribute in workspace that has 
```
REQUEST -->  to be inserted into db before you create a workspace 
CREATED -->  to be inserted into db after the workspace is created successfully
FAILED -->  to be inserted into db after the workspace is created successfully
```

similarly the fetch needs to change.